crates/block-producer/src/errors.rs

```diff
@@ -125,32 +127,32 @@ impl From<AddTransactionError> for tonic::Status {
 /// Error encountered while building a batch.
 #[derive(Debug, Error)]
 pub enum BuildBatchError {
-    #[error("Duplicated unauthenticated transaction input note ID in the batch: {0}")]
+    #[error("duplicated unauthenticated transaction input note ID in the batch: {0}")]
     DuplicateUnauthenticatedNote(NoteId),
 
-    #[error("Duplicated transaction output note ID in the batch: {0}")]
+    #[error("duplicated transaction output note ID in the batch: {0}")]
     DuplicateOutputNote(NoteId),
 
-    #[error("Note hashes mismatch for note {id}: (input: {input_hash}, output: {output_hash})")]
+    #[error("note hashes mismatch for note {id}: (input: {input_hash}, output: {output_hash})")]
     NoteHashesMismatch {
         id: NoteId,
         input_hash: Digest,
         output_hash: Digest,
     },
 
-    #[error("Failed to merge transaction delta into account {account_id}: {error}")]
+    #[error("failed to merge transaction delta into account {account_id}: {error}")]
     AccountUpdateError {
         account_id: AccountId,
         error: AccountDeltaError,
     },
 
-    #[error("Nothing actually went wrong, failure was injected on purpose")]
+    #[error("nothing actually went wrong, failure was injected on purpose")]
     InjectedFailure,
 
-    #[error("Batch proving task panic'd")]
+    #[error("batch proving task panic'd")]
     JoinError(#[from] tokio::task::JoinError),
 
-    #[error("Fetching inputs from store failed")]
+    #[error("fetching inputs from store failed")]
     StoreError(#[from] StoreError),
```

Comment:
```suggestion
    #[error("failed to fetch transaction batch inputs from store")]
    FailedToFetchBatchInputsFromStore(#[from] StoreError),
```

The error message is concrete, but the variant name is generic. Can we rename this? Feel free to pick something less unwiedly.

Also I think we should include in the error message what kinds of inputs we mean (there are also block inputs, and so it's not clear if "inputs" are those or tx inputs).

I would also prefer using `#[source]` on this one, so we are forced to map the error explicitly. Where we use this, we already use `map_err` anyway. Plus, this makes analyzing where variants are used possible in IDEs (i.e. find usages currently returns zero results for this because of the implicit nature of `#[from]`).